pkg/converter/converter_test.go

```diff
@@ -0,0 +1,196 @@
+	. "github.com/onsi/gomega"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	controlPlaneNamespace = "istio-system"
+	istioVersion          = supportedversion.Default
+	istioFile             = filepath.Join(project.RootDir, "tools", "istioConfig.yaml")
+	sailFile              = filepath.Join(project.RootDir, "tools", "istioConfig-sail.yaml")
+)
+
+func TestConversion(t *testing.T) {
+	testcases := []struct {
+		name           string
+		input          string
+		expectedOutput string
+		converterArgs  string
+	}{
+		{
+			name: "simple",
+			input: `apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: default
+spec:`,
+			expectedOutput: fmt.Sprintf(`apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: %s
+  namespace: %s`, istioVersion, controlPlaneNamespace),
+			converterArgs: fmt.Sprintf("%s %s -n %s -v %s", istioFile, sailFile, controlPlaneNamespace, istioVersion),
+		},
+		{
+			name: "complex",
+			input: `apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: default
+spec:
+  components:
+    base:
+      enabled: true
+    pilot:
+      enabled: false
+  values:
+    global:
+      externalIstiod: true
+      operatorManageWebhooks: true
+      configValidation: false
+    base:
+      enableCRDTemplates: true
+    pilot:
+      env:
+        PILOT_ENABLE_STATUS: true`,
+			expectedOutput: fmt.Sprintf(`apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  values:
+    global:
+      externalIstiod: true
+      operatorManageWebhooks: true
+      configValidation: false
+    base:
+      enableCRDTemplates: true
+      enabled: true
+    pilot:
+      env:
+        PILOT_ENABLE_STATUS: "true"
+      enabled: false
+  namespace: %s
+  version: %s`, controlPlaneNamespace, istioVersion),
+			converterArgs: fmt.Sprintf("%s %s -v %s -n %s", istioFile, sailFile, istioVersion, controlPlaneNamespace),
+		},
+		{
+			name: "mandatory-arguments-only",
+			input: `apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: default
+spec:`,
+			expectedOutput: fmt.Sprintf(`apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  namespace: %s`, controlPlaneNamespace),
+			converterArgs: istioFile,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Cleanup(func() {
+				os.Remove(istioFile)
+				os.Remove(sailFile)
+			})
+			g := NewWithT(t)
+			err := saveYamlToFile(tc.input, istioFile)
+			g.Expect(err).To(Succeed(), "failed to write YAML file")
+
+			g.Expect(executeConfigConverter(tc.converterArgs)).To(Succeed(), "error in execution of ./configuration-converter.sh")
+
+			isConversionValidated, err := validateYamlContent(sailFile, tc.expectedOutput)
+			g.Expect(err).To(Succeed(), fmt.Errorf("Error on validation: %w", err))
+			g.Expect(isConversionValidated).To(BeTrue(), "Converted content is not as expected")
```

Comment:
I believe this can be simplified to the following (the error is handled automatically).
```suggestion
			g.Expect(validateYamlContent(sailFile, tc.expectedOutput)).To(BeTrue(), "Converted content is not as expected")
```